internal/resource-request-operator/broadcaster.go

```diff
@@ -457,3 +459,21 @@ func getPodTransitionState(oldPod, newPod *corev1.Pod) PodTransition {
 
 	return PendingToPending
 }
+
+func isVirtualNode(node *corev1.Node) bool {
+	if virtualLabel, exists := node.Labels[consts.TypeLabel]; exists {
+		if virtualLabel == consts.TypeNode {
+			return true
+		}
+	}
+	return false
+}
+
+func isShadowPod(podToCheck *corev1.Pod) bool {
+	if shadowLabel, exists := podToCheck.Labels[consts.LocalPodLabelKey]; exists {
+		if shadowLabel == consts.LocalPodLabelValue {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
It may be worth to skip virtual nodes and shadow pods at the informer level (i.e. through a label selector), instead of manually here.